components/director/internal/domain/subscription/service.go

```diff
@@ -401,13 +417,43 @@ func (s *service) DetermineSubscriptionFlow(ctx context.Context, providerID, reg
 	return "", errors.Errorf("could not determine flow")
 }
 
-func (s *service) createApplicationFromTemplate(ctx context.Context, appTemplate *model.ApplicationTemplate, subscribedSubaccountID, consumerTenantID, subscribedAppName, subdomain, region string) error {
+func (s *service) createApplicationFromTemplate(ctx context.Context, appTemplate *model.ApplicationTemplate, subscribedSubaccountID, consumerTenantID, subscribedAppName, subdomain, region string, subscriptionPayload string) error {
 	values := []*model.ApplicationTemplateValueInput{
-		{Placeholder: "name", Value: subscribedAppName},
-		{Placeholder: "display-name", Value: subscribedAppName},
 		{Placeholder: "subdomain", Value: subdomain},
 		{Placeholder: "region", Value: strings.TrimPrefix(region, RegionPrefix)},
 	}
+
+	oldPlaceholders := appTemplate.Placeholders
```

Comment:
Can we extract the code from line 421 to 455 into a separate function - something like `preparePlaceholderValues` as currently it's a little hard to follow the parent `createApplicationFromTemplate` function flow because there are a lot of low level constructs like loops and re-assignments going on?